Terminal.Gui/Views/ColorPicker.cs

```diff
@@ -235,18 +247,22 @@ private void DisposeOldViews ()
         }
     }
 
-    private void RebuildColorFromBar (object? sender, EventArgs<int> e) { SetSelectedColor (_strategy.GetColorFromBars (_bars, Style.ColorModel), false); }
+    /// <exception cref="Exception">A delegate callback throws an exception.</exception>
+    private void RebuildColorFromBar (object? sender, EventArgs<int> e)
+    {
+        SetSelectedColor (_strategy.GetColorFromBars (_bars, Style.ColorModel), false);
+    }
 
+    /// <exception cref="Exception">A delegate callback throws an exception.</exception>
     private void SetSelectedColor (Color value, bool syncBars)
     {
         if (_selectedColor != value)
         {
-            Color old = _selectedColor;
+            RaisePropertyChanging (this, new (nameof (SelectedColor)));
             _selectedColor = value;
+            RaisePropertyChanged (this, new (nameof (SelectedColor)));
```

Comment:
I do not think we should start raising INPC until we have use-cases that require it (that won't work with the existing events). 

As I said here: https://github.com/gui-cs/Terminal.Gui/pull/3625#issuecomment-2265847460




> And regarding `INotifyPropertyChanged`: It may be used extensively, and it has some good characteristics, but it has some awful ones as well. It is designed primarily for data binding scenarios and (other than some awful things like requiring strings for property names) works great for that. I will not adopt it at the core of TG except in cases where the use case is primarily a data binding case.

> In the future, once we get around to building a XAML-based or -like markup UI system, it will be trivial to add `INotifyPropertyChanged` support on top of the pattern I am showing here. In fact, I think it could even be automated.

Reply:
Resolving this one too to keep the convo in one thread.